src/library_fs.js

```diff
@@ -1076,7 +1076,7 @@ FS.staticInit();
           }
         } else {
           // node doesn't exist, try to create it
-          node = FS.mknod(path, mode, 0);
+          node = FS.mknod(path, mode | 0o777, 0);
```

Comment:
Is there some reason why mknod with zero permissions doesn't work?

Reply:
The problem is later we try to open it and if we don't have read/write permissions that open will fail.

Reply:
Perhaps add a comment here.

```
// Ignore the permission bits here to ensure we can `open` this new file below.
// We use chmod below the apply the permissions once the file is open.
```

This makes me wonder if we should be able open and create in a single FS API call.. maybe our API has the wrong shape.  No need to fix as part of this PR.

Reply:
>  maybe our API has the wrong shape

I agree. I looked into this briefly, but it's confusing. Maybe the linux fs apis handle this better.